src/pkg/p2p/preheat/provider/preheat_image.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provider
+
+import (
+	"encoding/json"
+	"net/url"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// SupportedType indicates the supported preheating type 'image'.
+	SupportedType = "image"
+)
+
+// PreheatImage contains related information which can help providers to get/pull the images.
+type PreheatImage struct {
```

Comment:
PreheatImage is still in use? not Artifact?

Reply:
I change my mind. `Artifact` is too heavy for such a case and we need to do lots of changes. Preheat image is enough for such a case and we can avoid more changes and then save efforts.